pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,512 @@
+		skipPut          bool
+		expectedPutError error
+		expectedValue    *string
+	}{
+		"put var": {
+			env:   "AN_ENV",
+			value: ptrValue("value"),
+		},
+
+		"unset non-preset var": {
+			env:           "NEVER_SET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"unset preset var": {
+			presetValues:  map[string]string{"PRESET_ENV": "hey!"},
+			env:           "PRESET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"change preset var": {
+			presetValues:  map[string]string{"PRESET_ENV": "hey!"},
+			env:           "PRESET_ENV",
+			value:         ptrValue("hello!"),
+			expectedValue: ptrValue("hello!"),
+		},
+
+		"get unset env": {
+			skipPut:       true,
+			env:           "AN_UNSET_ENV",
+			expectedValue: ptrValue(""),
+		},
+
+		"put invalid env": {
+			env:              "",
+			value:            ptrValue("Invalid Value"),
+			expectedPutError: pam.BadItem,
+		},
+
+		"get invalid env": {
+			env:           "",
+			value:         ptrValue("Invalid Value"),
+			expectedValue: ptrValue(""),
+			skipPut:       true,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
+
+			require.False(t, tc.skipPut && tc.expectedValue == nil)
+
+			tx := NewModuleTransactionDummy(nil)
+			envList, err := tx.GetEnvList()
+			require.NoError(t, err)
+			require.Len(t, envList, 0)
+
+			if tc.presetValues != nil && !tc.skipPut {
+				for env, value := range tc.presetValues {
+					err := tx.PutEnv(env + "=" + value)
+					require.NoError(t, err)
+				}
+				envList, err = tx.GetEnvList()
+				require.NoError(t, err)
+				require.Equal(t, tc.presetValues, envList)
+			}
+
+			if !tc.skipPut {
+				var env string
+				if tc.value != nil {
+					env = tc.env + "=" + *tc.value
+				} else {
+					env = tc.env
+				}
+				err := tx.PutEnv(env)
+				require.ErrorIs(t, err, tc.expectedPutError)
+
+				if tc.expectedPutError == nil {
+					environment, err := tx.GetEnvList()
+					require.NoError(t, err)
+
+					if tc.value != nil && tc.expectedValue != nil {
+						require.Equal(t, map[string]string{
+							tc.env: *tc.expectedValue}, environment)
+					} else if tc.value != nil {
+						require.Equal(t, map[string]string{tc.env: *tc.value},
+							environment)
+					} else {
+						require.Equal(t, map[string]string{}, environment)
+					}
+				} else {
+					environment, err := tx.GetEnvList()
+					require.NoError(t, err)
+					require.Equal(t, map[string]string{}, environment)
+				}
+			}
+
+			if tc.expectedValue != nil {
+				value := tx.GetEnv(tc.env)
+				require.Equal(t, value, *tc.expectedValue)
+			}
+		})
+	}
+}
+
+func TestSetGetData(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		key              string
+		data             any
+		presetData       map[string]any
+		expectedSetError error
+		expectedGetError error
+		expectedData     any
+		skipSet          bool
+		skipGet          bool
+	}{
```

Comment:
${same comment about naming, ordering and blank line} 

(I won't repeat this on the other tests, but it applies as well :joy:)